File: src/test/java/io/veridia/jsonlogic/GeneralTests.java
package io.veridia.jsonlogic;

import com.fasterxml.jackson.core.JsonProcessingException;
import io.veridia.jsonlogic.helpers.ToBoolean;
import org.junit.Test;

import java.util.Collections;
import java.util.Map;

import static org.junit.Assert.*;

public class GeneralTests {
  private static final JsonLogic jsonLogic = new JsonLogic();

  @Test
  public void testNullExpression() throws JsonProcessingException {
      assertNull(jsonLogic.apply(null, null));
  }

  @Test
  public void testEmptyExpression() throws JsonProcessingException {
    var m = Map.of("one", 1);
    assertEquals(m, jsonLogic.apply("{\"var\": \"\"}", m));

    assertNull(jsonLogic.apply("{\"var\": \"\"}", null));

    assertNull(jsonLogic.apply("", null));
  }
}

File: src/test/scala/io/veridia/jsonlogic/GeneralTests.scala
package io.veridia.jsonlogic

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

import java.util as ju

class GeneralTests extends AnyFunSuite with Matchers:
  private val jsonLogic = new JsonLogic()

  test("null expression") {
    jsonLogic.apply(null, null) shouldBe null
  }

  test("empty expression") {
    val m = ju.Map.of("one", java.lang.Integer.valueOf(1))
    jsonLogic.apply("{\"var\": \"\"}", m) shouldEqual m
    jsonLogic.apply("{\"var\": \"\"}", null) shouldBe null
    jsonLogic.apply("", null) shouldBe null
  }
